src/border_agent/border_agent.cpp

```diff
@@ -210,6 +217,68 @@ void BorderAgent::Stop(void)
     UnpublishMeshCopService();
 }
 
+#if OTBR_ENABLE_EPSKC
+void BorderAgent::HandleEpskcStateChanged(void *aContext)
+{
+    BorderAgent *borderAgent = static_cast<BorderAgent *>(aContext);
+
+    if (otBorderAgentIsEphemeralKeyActive(borderAgent->mNcp.GetInstance()))
+    {
+        borderAgent->PublishEpskcService();
+    }
+    else
+    {
+        borderAgent->UnpublishEpskcService();
+    }
+}
+
+void BorderAgent::PublishEpskcService()
+{
+    otInstance *instance = mNcp.GetInstance();
+    int         port     = otBorderAgentGetUdpPort(instance);
+
+    otbrLogInfo("Publish meshcop-e service %s.%s.local. port %d", mServiceInstanceName.c_str(),
+                kBorderAgentEpskcServiceType, port);
+
+    mPublisher.PublishService(/* aHostName */ "", mServiceInstanceName, kBorderAgentEpskcServiceType,
+                              Mdns::Publisher::SubTypeList{}, port, {}, [this](otbrError aError) {
```

Comment:
nit: It's hard to infer what it is from a `{}`.  Please add an inline comment (or maybe add its type) indicating the meaning of the TXT data argument.

Reply:
Accepted; Thanks!